post-processor.go

```diff
@@ -161,14 +163,57 @@ func (p *PostProcessor) PostProcess(ui packer.Ui, artifact packer.Artifact) (pac
 	}
 	if size > 100*1024*1024 {
 		ui.Message("File size > 100MB. Initiating multipart upload")
-		multi, err := p.s3.Multi(boxPath, "application/octet-stream", "public-read")
+
+		multi, err := p.s3.InitMulti(boxPath, "application/octet-stream", "public-read")
 		if err != nil {
 			return nil, false, err
 		}
-		parts, err := multi.PutAll(file, 5*1024*1024)
-		if err != nil {
-			return nil, false, err
+
+		ui.Message("Uploading...")
+
+		const chunkSize = 10*1024*1024
+
+		totalParts := uint64(math.Ceil(float64(size) / float64(chunkSize)))
+		totalUploadSize := int64(0)
+
+		parts := []s3.Part{}
+
+		errorCount := 0;
+
+		for partNum := uint64(1); partNum <= totalParts; partNum++ {
+
+			filePos, err := file.Seek(0,1)
+			partSize := int(math.Min(chunkSize, float64(size - filePos)))
+			partBuffer := make([]byte, partSize)
+
+			ui.Message(fmt.Sprintf("Upload: Uploading part %d of %d, %d (of max %d) bytes", int(partNum), int(totalParts), int(partSize), int(chunkSize)))
+
+		  readBytes, err := file.Read(partBuffer)
+			ui.Message(fmt.Sprintf("Upload: Read %d bytes from box file on disk", readBytes))
+
+			bufferReader := bytes.NewReader(partBuffer)
```

Comment:
It would be easier to use an [`io.SectionReader`](https://golang.org/pkg/io/#SectionReader)